.github/workflows/main.yml

```diff
@@ -385,20 +338,30 @@ jobs:
   test_examples:
     needs:
     - build_examples
-    - build_nightly
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
     - uses: actions/download-artifact@v4
       with:
         name: examples1
         path: exbuild
+    - run: rustup update --no-self-update stable && rustup default stable
+    - run: cargo test -p example-tests
+      env:
+        EXBUILD: ${{ github.workspace }}/exbuild
+
+  test_nightly:
+    needs:
+    - build_nightly
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
     - uses: actions/download-artifact@v4
       with:
         name: examples2
         path: exbuild
     - run: rustup update --no-self-update stable && rustup default stable
-    - run: cargo test -p example-tests
+    - run: cargo test -p example-tests --no-default-features --features nightly
       env:
         EXBUILD: ${{ github.workspace }}/exbuild
```

Comment:
I think this would benefit from a matrix:
```yml
  test_examples:
    matrix:
    - needs: build_examples
      artifact: examples1
    - needs: build_nightly
      artifact: examples2
      args: --no-default-features --features nightly
    needs:
    - ${{ matrix.needs }}
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4
    - uses: actions/download-artifact@v4
      with:
        name: ${{ matrix.artifact }}
        path: exbuild
    - run: rustup update --no-self-update stable && rustup default stable
    - run: cargo test -p example-tests ${{ matrix.args }}
      env:
        EXBUILD: ${{ github.workspace }}/exbuild
```

Might not work because of `needs`.

Reply:
Doesn't work because of `needs` if I can trust the VSCode extension's validator.